dashboard/app/controllers/aichat_controller.rb

```diff
@@ -63,6 +63,32 @@ def log_chat_event
     render(status: :ok, json: response_body)
   end
 
+  # params are studentUserId: number, currentLevelId: number, scriptId: number
+  # POST /aichat/log_chat_event
+  def student_chat_history
+    # Request all chat events for a student at a given level/script.
+    begin
+      params.require([:studentUserId, :currentLevelId, :scriptId])
+    rescue ActionController::ParameterMissing
+      return render status: :bad_request, json: {}
+    end
+
+    # ensure that we have permission to view student's chat events, i.e.,
+    # student is in teacher section.
+    script_id = params[:scriptId]
+    level_id = params[:currentLevelId]
+    student_user_id = params[:studentUserId]
+    level = Level.find(level_id)
+    script_level = level.script_levels.find_by_script_id(script_id)
+    user = User.find(student_user_id)
+    unless can?(:view_as_user, script_level, user)
+      return render(status: :forbidden, json: {error: "Access denied for student chat history."})
+    end
+
+    chat_events = AichatEvent.where(user_id: student_user_id, level_id: level_id, script_id: script_id)
```

Comment:
Should we order these by most recent before sending back to the client (e.g. `.order(created_at: :desc)`)?

Also, do we need to return the whole model, or just the "aichat_event" (json) column?

Reply:
Oh yeah, just need to send `aichat_event`. And yes, should definitely order these. Thanks!